pkgs/development/interpreters/cbqn/default.nix

```diff
@@ -0,0 +1,88 @@
+let
+  # Allow the entire bootstrap sequence to be overridden
+  # if bootstrap == null, we just use the given bytecode and call it a day.
+  usingBytecode = generic:
+    generic {
+      useBytecode = true;
+      testSuitePasses = false;
+    };
+  bootstrapProcess = generic:
+    let
+      phase1 = usingBytecode generic;
+      phase2 = generic { bqn = phase1; };
+      phase3 = generic { bqn = phase2; };
+    in phase3;
+in { stdenv, lib, fetchFromGitHub, bootstrap ? bootstrapProcess, ripgrep }:
+
+let
+  libBQN = fetchFromGitHub {
+    owner = "mlochbaum";
+    repo = "bqn";
+    rev = "e46d1edb07cea7f9a5be26165213db8a17b4fa4f";
+    sha256 = "13kbak06ivw935jfdc98k8bxy0qkpdj2xwp1ajqbkckbr02pqz40";
+  };
+
+  bytecode = fetchFromGitHub {
+    owner = "dzaima";
+    repo = "cbqn";
+    rev = "6aa6834666e91f2490a86efd249ba9c7b1d35794";
+    sha256 = "046k0gkwc5mw4wm551hsgiqcvmh0a7j43pr55l8sh6faliqgrnhi";
+  };
+
+  src = fetchFromGitHub {
+    owner = "dzaima";
+    repo = "cbqn";
+    rev = "a7ee8044d165119c6df8e93c5c13286bac54bf59";
+    sha256 = "1zqfa8pzqhgg1n7xs46fjqhqh3njsyrk9b8lbzcl6ka4rrjkx4j5";
+  };
+
+  generic = { useBytecode ? false, bqn ? null, testSuitePasses ? true }:
+    # Either use bytecode, or have bqn specified.
+    assert (bqn != null) == !useBytecode;
+
+    stdenv.mkDerivation rec {
+      pname = "cbqn" + lib.optionalString useBytecode "-bytecode";
+      version = "unstable-2021-08-24";
+
+      inherit src;
+      patches = lib.optional useBytecode ./generated.patch;
```

Comment:
I'd apply that unconditionally if possible. Simpler and makes sure it always breaks when the patch becomes stale (maybe we'll have a bootstrap method without bytecode later…).